### PR TITLE
chore: Bump OpenDAL to v0.36

### PR DESCRIPTION
**Pull request by Xuanwo**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

This PR will bump OpenDAL to v0.36

All changes could be found here: https://opendal.apache.org/docs/rust/opendal/docs/changelog/index.html

## Checklist

- [x]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)

Related to https://github.com/GreptimeTeam/greptimedb/pull/1677

### CODE REVIEW

**Comment by Xuanwo** [comment]
Would @waynexia be willing to review this PR?

File: src/datanode/src/store.rs [hunk 1]
```diff
@@ -56,7 +56,8 @@ pub(crate) async fn new_object_store(store_config: &ObjectStoreConfig) -> Result
             LoggingLayer::default()
                 // Print the expected error only in DEBUG level.
                 // See https://docs.rs/opendal/latest/opendal/layers/struct.LoggingLayer.html#method.with_error_level
-                .with_error_level(Some(log::Level::Debug)),
+                .with_error_level(Some("debug"))
```
Comment (killme2008) [inline src/datanode/src/store.rs:59]: Cool, can you remove `log = "0.4"` in `src/datanode/Cargo.toml` too?

**Review by killme2008 — APPROVED** [review]
LGTM

**Review by waynexia — APPROVED** [review]
LGTM, thanks 🚀

File: src/datanode/src/store.rs (hunk 1 above)
Comment (Xuanwo) [reply src/datanode/src/store.rs:59]: Addressed.

**Review by v0y4g3r — APPROVED** [review]
LGTM.

**Pull request merged**